crates/core_arch/src/wasm/mod.rs

```diff
@@ -0,0 +1,127 @@
+//! operate on the `v128` type to perform data processing. The SIMD proposal
+//! at the time of this writing is in [phase 4] which means that it's in the
+//! standardization phase. It's expected that once some testing on nightly
+//! has happened a stabilization proposal will be made for the Rust
+//! intrinsics. If you notice anything awry please feel free to [open an
+//! issue](https://github.com/rust-lang/stdarch/issues/new).
+//!
+//! [phase 4]: https://github.com/webassembly/proposals
+//!
+//! Using SIMD is intended to be similar to as you would on `x86_64`, for
+//! example. You'd write a function such as:
+//!
+//! ```rust,ignore
+//! #[target_feature(enable = "simd128")]
+//! unsafe fn uses_simd() {
+//!     #[cfg(target_arch = "wasm32")]
+//!     use std::arch::wasm32::*;
+//!     #[cfg(target_arch = "wasm64")]
+//!     use std::arch::wasm64::*;
+//!     // ...
+//! }
+//! ```
+//!
+//! Unlike `x86_64`, however, WebAssembly does not currently have dynamic
+//! detection at runtime as to whether SIMD is supported (this is one of the
+//! motivators for the [conditional sections][condsections] and [feature
+//! detection] proposals, but that is still pretty early days). This means
+//! that your binary will either have SIMD and can only run on engines
+//! which support SIMD, or it will not have SIMD at all. For compatibility
+//! the standard library itself does not use any SIMD internally.
+//! Determining how best to ship your WebAssembly binary with SIMD is
+//! largely left up to you as it can can be pretty nuanced depending on
+//! your situation.
+//!
+//! [condsections]: https://github.com/webassembly/conditional-sections
+//! [feature detection]: https://github.com/WebAssembly/feature-detection
+//!
+//! To enable SIMD support at compile time you need to do one of two things:
+//!
+//! * First you can annotate functions with `#[target_feature(enable =
+//!   "simd128")]`. This causes just that one function to have SIMD support
+//!   available to it, and intrinsics will get inlined as usual in this
+//!   situation.
+//!
+//! * Second you can compile your program with `-Ctarget-feature=+simd128`.
+//!   This compilation flag blanket enables SIMD support for your entire
+//!   compilation. Note that this does not include the standard library
+//!   unless you [recompile the standard library][buildstd].
+//!
+//! [buildstd]: https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#build-std
+//!
+//! If you enable SIMD via either of these routes then you'll have a
+//! WebAssembly binary that uses SIMD instructions, and you'll need to ship
+//! that accordingly. Also note that if you call SIMD intrinsics but don't
+//! enable SIMD via either of these mechanisms, you'll still have SIMD
+//! generated in your program. This means to generate a binary without SIMD
+//! you'll need to avoid both options above plus calling into any intrinsics
+//! in this module.
```

Comment:
This module is private so these docs are not actually visible in the documentation.